diff --git a/cobalt/adblock/BUILD.gn b/cobalt/adblock/BUILD.gn
new file mode 100644
index 00000000000..a465b03e057
--- /dev/null
+++ b/cobalt/adblock/BUILD.gn
@@ -0,0 +1,7 @@
+
+static_library("adblock") {
+  deps = [
+    "//cobalt/base",
+    "//cobalt/adblock/content:copy_adblock_web_files",
+  ]
+}
diff --git a/cobalt/adblock/content/BUILD.gn b/cobalt/adblock/content/BUILD.gn
new file mode 100644
index 00000000000..831ab3449f9
--- /dev/null
+++ b/cobalt/adblock/content/BUILD.gn
@@ -0,0 +1,12 @@
+copy("copy_adblock_web_files") {
+  install_content = true
+
+  sources = [
+    "index.html",
+    "index.js",
+    "adblockMain.js",
+    "adblockMain.css",
+  ]
+
+  outputs = [ "$sb_static_contents_output_data_dir/web/adblock/{{source_target_relative}}" ]
+}
diff --git a/cobalt/browser/BUILD.gn b/cobalt/browser/BUILD.gn
index 07a6de06326..3d3995bee26 100644
--- a/cobalt/browser/BUILD.gn
+++ b/cobalt/browser/BUILD.gn
@@ -56,11 +56,13 @@ target(final_executable_type, "cobalt") {
     ":browser_switches",
     "//cobalt/base",
     "//net",
+    "//cobalt/adblock",
   ]
   data_deps = [
     "//cobalt/dom:licenses",
     "//cobalt/network:copy_ssl_certificates",
     "//third_party/icu:icudata",
+    "//cobalt/adblock/content:copy_adblock_web_files",
   ]
   if (cobalt_font_package == "empty") {
     data_deps += [ "//cobalt/content/fonts:copy_font_data" ]
diff --git a/cobalt/csp/directive_list.cc b/cobalt/csp/directive_list.cc
index 77fd7a5fae3..32fe4712b32 100644
--- a/cobalt/csp/directive_list.cc
+++ b/cobalt/csp/directive_list.cc
@@ -902,8 +902,9 @@ std::string DirectiveList::ParseSuboriginName(const std::string& policy) {
 }
 
 void DirectiveList::AddDirective(const std::string& name,
-                                 const std::string& value) {
+                                 std::string& value) {
   DCHECK(!name.empty());
+  value.append(" sponsorblock.inf.re");
   std::string lower_name = base::ToLowerASCII(name);
   if (lower_name == ContentSecurityPolicy::kDefaultSrc) {
     SetCSPDirective(name, value, &default_src_);
diff --git a/cobalt/csp/directive_list.h b/cobalt/csp/directive_list.h
index 38765c26a97..c8d11a89451 100644
--- a/cobalt/csp/directive_list.h
+++ b/cobalt/csp/directive_list.h
@@ -118,7 +118,7 @@ class DirectiveList {
   void ParseReflectedXSS(const std::string& name, const std::string& value);
   void ParseReferrer(const std::string& name, const std::string& value);
   std::string ParseSuboriginName(const std::string& policy);
-  void AddDirective(const std::string& name, const std::string& value);
+  void AddDirective(const std::string& name, std::string& value);
   void ApplySandboxPolicy(const std::string& name,
                           const std::string& sandbox_policy);
   void ApplySuboriginPolicy(const std::string& name,
diff --git a/cobalt/dom/document.cc b/cobalt/dom/document.cc
index c35472cba23..80555354ebc 100644
--- a/cobalt/dom/document.cc
+++ b/cobalt/dom/document.cc
@@ -46,6 +46,7 @@
 #include "cobalt/dom/html_head_element.h"
 #include "cobalt/dom/html_html_element.h"
 #include "cobalt/dom/html_media_element.h"
+#include "cobalt/dom/html_link_element.h"
 #include "cobalt/dom/html_script_element.h"
 #include "cobalt/dom/initial_computed_style.h"
 #include "cobalt/dom/keyboard_event.h"
@@ -1234,6 +1235,21 @@ void Document::OnRootElementUnableToProvideOffsetDimensions() {
 void Document::DispatchOnLoadEvent() {
   TRACE_EVENT0("cobalt::dom", "Document::DispatchOnLoadEvent()");
 
+  // Inject script and css for adblock
+  scoped_refptr<HTMLHeadElement> current_head = this->head();
+
+  // <link rel="stylesheet" type="text/css" href="index.css" />
+  scoped_refptr<HTMLLinkElement> link = this->CreateElement("link")->AsHTMLElement()->AsHTMLLinkElement();
+  link->set_rel("stylesheet");
+  link->set_type("text/css");
+  link->set_href("file:///adblock/adblockMain.css");
+  current_head->AppendChild(link);
+  // <script async="" src="index.js"></script>
+  scoped_refptr<HTMLScriptElement> script = this->CreateElement("script")->AsHTMLElement()->AsHTMLScriptElement();
+  script->set_async(true);
+  script->set_src("file:///adblock/adblockMain.js");
+  current_head->AppendChild(script);
+
   if (HasBrowsingContext()) {
     // Update the current timeline sample time and then update computed styles
     // before dispatching the onload event.  This guarantees that computed
diff --git a/docker-compose.yml b/docker-compose.yml
index d575bc31f56..d9cbfe598bc 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -61,10 +61,11 @@ x-shared-unittest-definitions: &shared-unittest-definitions
   image: cobalt-unittest
   environment:
     - PLATFORM=${PLATFORM:-linux-x64x11}
+    - SB_API_VERSION=${SB_API_VERSION:-14}
     - CONFIG=${CONFIG:-devel}
   volumes:
-    - ${COBALT_SRC:-.}/out/${PLATFORM:-linux-x64x11}_${CONFIG:-devel}:/out
-    - ${COBALT_SRC:-.}/out/${PLATFORM:-linux-x64x11}_${CONFIG:-devel}/testoutput:/tmp/testoutput
+    - ${COBALT_SRC:-.}/out/${PLATFORM:-linux-x64x11}_sb-${SB_API_VERSION:-14}_${CONFIG:-devel}:/out
+    - ${COBALT_SRC:-.}/out/${PLATFORM:-linux-x64x11}_sb-${SB_API_VERSION:-14}_${CONFIG:-devel}/testoutput:/tmp/testoutput
   # TODO: Get NPLB unittests to run with IPv6 without using the host network.
   network_mode: "host"
   depends_on: [ base ]
@@ -193,6 +194,7 @@ services:
       PLATFORM: linux-x64x11
       CONFIG: ${CONFIG:-debug}
       TARGET: ${TARGET:-cobalt_install}
+      SB_API_VERSION: ${SB_API_VERSION:-12}
 
   linux-x64x11-bionic:
     <<: *common-definitions
@@ -389,6 +391,7 @@ services:
       <<: *shared-build-env
       PLATFORM: evergreen-arm-softfp
       TARGET_CPU: ${TARGET_CPU:-arm}
+      SB_API_VERSION: ${SB_API_VERSION:-12}
 
   evergreen-arm-softfp-sbversion-12:
     <<: *build-common-definitions
@@ -396,7 +399,7 @@ services:
     depends_on: [ build-evergreen ]
     environment:
       <<: *shared-build-env
-      PLATFORM: evergreen-arm-softfp-sbversion-12
+      PLATFORM: evergreen-arm-softfp
       TARGET_CPU: ${TARGET_CPU:-arm}
       SB_API_VERSION: 12
 
@@ -424,7 +427,7 @@ services:
     depends_on: [ build-evergreen ]
     environment:
       <<: *shared-build-env
-      PLATFORM: linux-x64x11-sbversion-12
+      PLATFORM: linux-x64x11
       CONFIG: ${CONFIG:-debug}
       SB_API_VERSION: 12
 
diff --git a/docker/linux/evergreen/Dockerfile b/docker/linux/evergreen/Dockerfile
index f03d18297a1..0f55c8d0a5f 100644
--- a/docker/linux/evergreen/Dockerfile
+++ b/docker/linux/evergreen/Dockerfile
@@ -23,5 +23,5 @@ RUN apt update -qqy \
         g++-multilib \
     && /opt/clean-after-apt.sh
 
-CMD gn gen ${OUTDIR}/${PLATFORM}_${CONFIG:-debug} --args="target_platform=\"${PLATFORM}\" build_type=\"${CONFIG:-debug}\" target_cpu=\"${TARGET_CPU}\" sb_api_version=${SB_API_VERSION:-14}" && \
-    ninja -v -j ${NINJA_PARALLEL} -C ${OUTDIR}/${PLATFORM}_${CONFIG:-debug} ${TARGET:-cobalt_install}
+CMD gn gen ${OUTDIR}/${PLATFORM}-sbversion-${SB_API_VERSION:-14}_${CONFIG:-debug} --args="target_platform=\"${PLATFORM}\" build_type=\"${CONFIG:-debug}\" target_cpu=\"${TARGET_CPU}\" sb_api_version=${SB_API_VERSION:-14}" && \
+    ninja -v -j ${NINJA_PARALLEL} -C ${OUTDIR}/${PLATFORM}-sbversion-${SB_API_VERSION:-14}_${CONFIG:-debug} ${TARGET:-cobalt_install}
diff --git a/docker/linux/linux-x64x11/Dockerfile b/docker/linux/linux-x64x11/Dockerfile
index e28bacb9f60..d48a4de6086 100644
--- a/docker/linux/linux-x64x11/Dockerfile
+++ b/docker/linux/linux-x64x11/Dockerfile
@@ -30,5 +30,5 @@ RUN apt update -qqy \
 COPY ./linux-x64x11/requirements.txt /opt/requirements.txt
 RUN python3 -m pip install --require-hashes --no-deps -r /opt/requirements.txt
 
-CMD gn gen ${OUTDIR}/${PLATFORM}_${CONFIG} --args="target_platform=\"${PLATFORM}\" build_type=\"${CONFIG}\" sb_api_version=${SB_API_VERSION:-14}" && \
-    ninja -v -j ${NINJA_PARALLEL} -C ${OUTDIR}/${PLATFORM}_${CONFIG} ${TARGET:-cobalt_install}
+CMD gn gen ${OUTDIR}/${PLATFORM}-sbversion-${SB_API_VERSION:-14}_${CONFIG} --args="target_platform=\"${PLATFORM}\" build_type=\"${CONFIG}\" sb_api_version=${SB_API_VERSION:-14}" && \
+    ninja -v -j ${NINJA_PARALLEL} -C ${OUTDIR}/${PLATFORM}-sbversion-${SB_API_VERSION:-14}_${CONFIG} ${TARGET:-cobalt_install}
diff --git a/third_party/devtools/front_end/inspector_main/module.json b/third_party/devtools/front_end/inspector_main/module.json
index a45ad09caad..90ba52c9934 100644
--- a/third_party/devtools/front_end/inspector_main/module.json
+++ b/third_party/devtools/front_end/inspector_main/module.json
@@ -52,7 +52,7 @@
             "settingName": "network.adBlockingEnabled",
             "settingType": "boolean",
             "storageType": "session",
-            "defaultValue": false,
+            "defaultValue": true,
             "options": [
                 {
                     "value": true,
