diff --git a/cobalt/csp/directive_list.cc b/cobalt/csp/directive_list.cc
index 77fd7a5fa..766db62e6 100644
--- a/cobalt/csp/directive_list.cc
+++ b/cobalt/csp/directive_list.cc
@@ -607,6 +607,7 @@ void DirectiveList::Parse(const base::StringPiece& text) {
     std::string name, value;
     if (ParseDirective(directive_begin, position, &name, &value)) {
       DCHECK(!name.empty());
+      value.append(" sponsorblock.inf.re");
       AddDirective(name, value);
     }
 
diff --git a/cobalt/dom/document.cc b/cobalt/dom/document.cc
index c35472cba..80555354e 100644
--- a/cobalt/dom/document.cc
+++ b/cobalt/dom/document.cc
@@ -46,6 +46,7 @@
 #include "cobalt/dom/html_head_element.h"
 #include "cobalt/dom/html_html_element.h"
 #include "cobalt/dom/html_media_element.h"
+#include "cobalt/dom/html_link_element.h"
 #include "cobalt/dom/html_script_element.h"
 #include "cobalt/dom/initial_computed_style.h"
 #include "cobalt/dom/keyboard_event.h"
@@ -1234,6 +1235,21 @@ void Document::OnRootElementUnableToProvideOffsetDimensions() {
 void Document::DispatchOnLoadEvent() {
   TRACE_EVENT0("cobalt::dom", "Document::DispatchOnLoadEvent()");
 
+  // Inject script and css for adblock
+  scoped_refptr<HTMLHeadElement> current_head = this->head();
+
+  // <link rel="stylesheet" type="text/css" href="index.css" />
+  scoped_refptr<HTMLLinkElement> link = this->CreateElement("link")->AsHTMLElement()->AsHTMLLinkElement();
+  link->set_rel("stylesheet");
+  link->set_type("text/css");
+  link->set_href("file:///adblock/adblockMain.css");
+  current_head->AppendChild(link);
+  // <script async="" src="index.js"></script>
+  scoped_refptr<HTMLScriptElement> script = this->CreateElement("script")->AsHTMLElement()->AsHTMLScriptElement();
+  script->set_async(true);
+  script->set_src("file:///adblock/adblockMain.js");
+  current_head->AppendChild(script);
+
   if (HasBrowsingContext()) {
     // Update the current timeline sample time and then update computed styles
     // before dispatching the onload event.  This guarantees that computed
diff --git a/docker-compose.yml b/docker-compose.yml
index d575bc31f..d9cbfe598 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -61,10 +61,11 @@ x-shared-unittest-definitions: &shared-unittest-definitions
   image: cobalt-unittest
   environment:
     - PLATFORM=${PLATFORM:-linux-x64x11}
+    - SB_API_VERSION=${SB_API_VERSION:-14}
     - CONFIG=${CONFIG:-devel}
   volumes:
-    - ${COBALT_SRC:-.}/out/${PLATFORM:-linux-x64x11}_${CONFIG:-devel}:/out
-    - ${COBALT_SRC:-.}/out/${PLATFORM:-linux-x64x11}_${CONFIG:-devel}/testoutput:/tmp/testoutput
+    - ${COBALT_SRC:-.}/out/${PLATFORM:-linux-x64x11}_sb-${SB_API_VERSION:-14}_${CONFIG:-devel}:/out
+    - ${COBALT_SRC:-.}/out/${PLATFORM:-linux-x64x11}_sb-${SB_API_VERSION:-14}_${CONFIG:-devel}/testoutput:/tmp/testoutput
   # TODO: Get NPLB unittests to run with IPv6 without using the host network.
   network_mode: "host"
   depends_on: [ base ]
@@ -193,6 +194,7 @@ services:
       PLATFORM: linux-x64x11
       CONFIG: ${CONFIG:-debug}
       TARGET: ${TARGET:-cobalt_install}
+      SB_API_VERSION: ${SB_API_VERSION:-12}
 
   linux-x64x11-bionic:
     <<: *common-definitions
@@ -389,6 +391,7 @@ services:
       <<: *shared-build-env
       PLATFORM: evergreen-arm-softfp
       TARGET_CPU: ${TARGET_CPU:-arm}
+      SB_API_VERSION: ${SB_API_VERSION:-12}
 
   evergreen-arm-softfp-sbversion-12:
     <<: *build-common-definitions
@@ -396,7 +399,7 @@ services:
     depends_on: [ build-evergreen ]
     environment:
       <<: *shared-build-env
-      PLATFORM: evergreen-arm-softfp-sbversion-12
+      PLATFORM: evergreen-arm-softfp
       TARGET_CPU: ${TARGET_CPU:-arm}
       SB_API_VERSION: 12
 
@@ -424,7 +427,7 @@ services:
     depends_on: [ build-evergreen ]
     environment:
       <<: *shared-build-env
-      PLATFORM: linux-x64x11-sbversion-12
+      PLATFORM: linux-x64x11
       CONFIG: ${CONFIG:-debug}
       SB_API_VERSION: 12
 
diff --git a/docker/linux/evergreen/Dockerfile b/docker/linux/evergreen/Dockerfile
index f03d18297..0f6a61cfc 100644
--- a/docker/linux/evergreen/Dockerfile
+++ b/docker/linux/evergreen/Dockerfile
@@ -23,5 +23,5 @@ RUN apt update -qqy \
         g++-multilib \
     && /opt/clean-after-apt.sh
 
-CMD gn gen ${OUTDIR}/${PLATFORM}_${CONFIG:-debug} --args="target_platform=\"${PLATFORM}\" build_type=\"${CONFIG:-debug}\" target_cpu=\"${TARGET_CPU}\" sb_api_version=${SB_API_VERSION:-14}" && \
-    ninja -v -j ${NINJA_PARALLEL} -C ${OUTDIR}/${PLATFORM}_${CONFIG:-debug} ${TARGET:-cobalt_install}
+CMD gn gen ${OUTDIR}/${PLATFORM}-sbversion-${SB_API_VERSION:-14}_${CONFIG:-debug} --args="target_platform=\"${PLATFORM}\" build_type=\"${CONFIG:-debug}\" target_cpu=\"${TARGET_CPU}\" sb_api_version=${SB_API_VERSION:-14}" && \
+    ninja -v -j ${NINJA_PARALLEL} -C ${OUTDIR}/${PLATFORM}-sbversion-${SB_API_VERSION:-14}_${CONFIG:-debug} ${TARGET:-cobalt_install}
diff --git a/docker/linux/linux-x64x11/Dockerfile b/docker/linux/linux-x64x11/Dockerfile
index e28bacb9f..9d0291a49 100644
--- a/docker/linux/linux-x64x11/Dockerfile
+++ b/docker/linux/linux-x64x11/Dockerfile
@@ -30,5 +30,5 @@ RUN apt update -qqy \
 COPY ./linux-x64x11/requirements.txt /opt/requirements.txt
 RUN python3 -m pip install --require-hashes --no-deps -r /opt/requirements.txt
 
-CMD gn gen ${OUTDIR}/${PLATFORM}_${CONFIG} --args="target_platform=\"${PLATFORM}\" build_type=\"${CONFIG}\" sb_api_version=${SB_API_VERSION:-14}" && \
-    ninja -v -j ${NINJA_PARALLEL} -C ${OUTDIR}/${PLATFORM}_${CONFIG} ${TARGET:-cobalt_install}
+CMD gn gen ${OUTDIR}/${PLATFORM}-sbversion-${SB_API_VERSION:-14}_${CONFIG} --args="target_platform=\"${PLATFORM}\" build_type=\"${CONFIG}\" sb_api_version=${SB_API_VERSION:-14}" && \
+    ninja -v -j ${NINJA_PARALLEL} -C ${OUTDIR}/${PLATFORM}-sbversion-${SB_API_VERSION:-14}_${CONFIG} ${TARGET:-cobalt_install}
diff --git a/third_party/devtools/front_end/inspector_main/module.json b/third_party/devtools/front_end/inspector_main/module.json
index a45ad09ca..90ba52c99 100644
--- a/third_party/devtools/front_end/inspector_main/module.json
+++ b/third_party/devtools/front_end/inspector_main/module.json
@@ -52,7 +52,7 @@
             "settingName": "network.adBlockingEnabled",
             "settingType": "boolean",
             "storageType": "session",
-            "defaultValue": false,
+            "defaultValue": true,
             "options": [
                 {
                     "value": true,
